src/oscar/apps/basket/abstract_models.py

```diff
@@ -365,8 +365,9 @@ def _create_line_reference(self, product, stockrecord, options):
         base = '%s_%s' % (product.id, stockrecord.id)
         if not options:
             return base
-        repr_options = [{'option': repr(option['option']),
-                         'value': repr(option['value'])} for option in options]
+        repr_options = [(repr(option['option']), repr(option['value']))
```

Comment:
I'd prefer not to change the content of what is being hashed, to reduce the risk of breaking an existing hash. I think we can make the existing hash values deterministic by doing this instead:

```
repr_options.sort(key=lambda a: a['option'])
```